src/main/java/com/budiyev/android/codescanner/Utils.java

```diff
@@ -188,12 +188,9 @@ public static Point getPreviewSize(int imageWidth, int imageHeight, int frameWid
         if (imageWidth == frameWidth && imageHeight == frameHeight) {
             return new Point(frameWidth, frameHeight);
         }
-        int imageDivisor = greatestCommonDivisor(imageWidth, imageHeight);
-        int imageRatioWidth = imageWidth / imageDivisor;
-        int imageRatioHeight = imageHeight / imageDivisor;
-        int resultWidth = imageRatioWidth * frameHeight / imageRatioHeight;
+        int resultWidth = imageWidth * frameHeight / imageHeight;
         if (resultWidth < frameWidth) {
-            return new Point(frameWidth, imageRatioHeight * frameWidth / imageRatioWidth);
+            return new Point(frameWidth, imageWidth * frameWidth / imageHeight);
```

Comment:
I think this line is wrong, should be imageHeight * frameWidth / imageWidth

Reply:
Test with the following layout:

    <RelativeLayout
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:background="#ff00ff00">

        <com.budiyev.android.codescanner.CodeScannerView
            android:id="@+id/scanner_view"
            android:layout_width="match_parent"
            android:layout_height="250dp"
            app:autoFocusButtonColor="@android:color/white"
            app:autoFocusButtonVisible="true"
            app:flashButtonColor="@android:color/white"
            android:layout_centerVertical="true"
            app:flashButtonVisible="true"
            app:frameColor="@android:color/white"
            app:frameCornersSize="50dp"
            app:frameAspectRatioWidth="1"
            app:frameAspectRatioHeight="1"
            app:frameSize="0.75"
            app:frameThickness="2dp"
            app:maskColor="#77000000"/>
    </RelativeLayout>

Using your version, the preview uses the full height. Using my version, it uses about the space it should use. The frame has always the correct size, but the preview did not.